optuna/visualization/_timeline.py

```diff
@@ -0,0 +1,134 @@
+import datetime
+from typing import List
+from typing import NamedTuple
+
+from optuna._experimental import experimental_func
+from optuna.logging import get_logger
+from optuna.study import Study
+from optuna.trial import FrozenTrial
+from optuna.trial import TrialState
+from optuna.visualization._pareto_front import _make_hovertext
```

Comment:
It is not good practice to rely on private functions for other visualization functions. If you want to use this function, why not move it to `optuna.visualization._utils`?

Reply:
Agreed. I moved `_make_hovertext` to `optuna.visualization._utils`. This required some additional changes of existing source code.